diff --git i/pkg/resolution/resolver/git/repository.go w/pkg/resolution/resolver/git/repository.go
index 66ee335d3..d015ba7c2 100644
--- i/pkg/resolution/resolver/git/repository.go
+++ w/pkg/resolution/resolver/git/repository.go
@@ -120,7 +120,7 @@ func (repo *repository) execGit(ctx context.Context, subCmd string, args ...stri
 		}
 	}
 	cmd := repo.executor(ctx, "git", append(configArgs, args...)...)
-	cmd.Env = append(cmd.Env, env...)
+	cmd.Env = append(cmd.Environ(), env...)
 
 	out, err := cmd.Output()
 	if err != nil {
